src/mon/OSDMonitor.cc

```diff
@@ -4551,6 +4551,7 @@ int OSDMonitor::crush_rename_bucket(const string& srcname,
     return ret;
 
   pending_inc.crush.clear();
+  newcrush.cleanup_classes();
   newcrush.encode(pending_inc.crush, mon->get_quorum_con_features());
```

Comment:
These seem very fragile because the map is unusable after this.  (Same with decompile_crush(), although there it's rebuilt so just wasted work).  Perhaps instead it's better to make encode smart enough to skip all the extra buckets?